pkg/lib/queueinformer/queueinformer_operator.go

```diff
@@ -133,7 +136,9 @@ func (o *Operator) sync(loop *QueueInformer, key string) error {
 	if err != nil {
 		return err
 	}
-
+	if err = o.handleMetrics(loop); err != nil {
```

Comment:
if sync gets retried because syncHandler returns false (see `processNextWorkItem`) this will get called more than we really want. 

Would it make sense to trigger only after successful `syncHandler`? e.g. 

```go
success := loop.syncHandler()
if success {
  go o.handleMetrics(loop) // are prom metrics safe to call concurrently?
}
return success
```